pkg/cmd/openshift-apiserver/openshiftapiserver/config.go

```diff
@@ -0,0 +1,294 @@
+		return nil, err
+	}
+
+	genericConfig := genericapiserver.NewRecommendedConfig(legacyscheme.Codecs)
+	// Current default values
+	//Serializer:                   codecs,
+	//ReadWritePort:                443,
+	//BuildHandlerChainFunc:        DefaultBuildHandlerChain,
+	//HandlerChainWaitGroup:        new(utilwaitgroup.SafeWaitGroup),
+	//LegacyAPIGroupPrefixes:       sets.NewString(DefaultLegacyAPIPrefix),
+	//DisabledPostStartHooks:       sets.NewString(),
+	//HealthzChecks:                []healthz.HealthzChecker{healthz.PingHealthz, healthz.LogHealthz},
+	//EnableIndex:                  true,
+	//EnableDiscovery:              true,
+	//EnableProfiling:              true,
+	//EnableMetrics:                true,
+	//MaxRequestsInFlight:          400,
+	//MaxMutatingRequestsInFlight:  200,
+	//RequestTimeout:               time.Duration(60) * time.Second,
+	//MinRequestTimeout:            1800,
+	//EnableAPIResponseCompression: utilfeature.DefaultFeatureGate.Enabled(features.APIResponseCompression),
+	//LongRunningFunc: genericfilters.BasicLongRunningRequestCheck(sets.NewString("watch"), sets.NewString()),
+
+	// TODO this is actually specific to the kubeapiserver
+	//RuleResolver authorizer.RuleResolver
+	genericConfig.SharedInformerFactory = kubeInformers
+	genericConfig.ClientConfig = kubeClientConfig
+
+	// these are set via options
+	//SecureServing *SecureServingInfo
+	//Authentication AuthenticationInfo
+	//Authorization AuthorizationInfo
+	//LoopbackClientConfig *restclient.Config
+	// this is set after the options are overlayed to get the authorizer we need.
+	//AdmissionControl      admission.Interface
+	//ReadWritePort int
+	//PublicAddress net.IP
+
+	// these are defaulted sanely during complete
+	//DiscoveryAddresses discovery.Addresses
+
+	genericConfig.CorsAllowedOriginList = openshiftAPIServerConfig.CORSAllowedOrigins
+	genericConfig.Version = &openshiftVersion
+	// we don't use legacy audit anymore
+	genericConfig.LegacyAuditWriter = nil
+	genericConfig.AuditBackend = backend
+	genericConfig.AuditPolicyChecker = policyChecker
+	genericConfig.ExternalAddress = "apiserver.openshift-apiserver.svc"
+	genericConfig.BuildHandlerChainFunc = OpenshiftHandlerChain
+	genericConfig.LegacyAPIGroupPrefixes = configprocessing.LegacyAPIGroupPrefixes
+	genericConfig.RequestInfoResolver = configprocessing.OpenshiftRequestInfoResolver()
+	genericConfig.OpenAPIConfig = configprocessing.DefaultOpenAPIConfig(*openshiftAPIServerConfig)
+	genericConfig.SwaggerConfig = defaultSwaggerConfig()
+	genericConfig.RESTOptionsGetter = restOptsGetter
+	// previously overwritten.  I don't know why
+	genericConfig.RequestTimeout = time.Duration(60) * time.Second
+	genericConfig.MinRequestTimeout = openshiftAPIServerConfig.ServingInfo.RequestTimeoutSeconds
+	genericConfig.MaxRequestsInFlight = openshiftAPIServerConfig.ServingInfo.MaxRequestsInFlight
+	genericConfig.MaxMutatingRequestsInFlight = openshiftAPIServerConfig.ServingInfo.MaxRequestsInFlight / 2
```

Comment:
do we have all this logic twice now?

Reply:
> do we have all this logic twice now?

Yes, but I'd rather not touch the other since it's more closely to tied to kube and in 4.0 I want to try to skin it a different way.